src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs

```diff
@@ -38,14 +43,18 @@ public class PackageSource : IEquatable<PackageSource>
         /// </summary>
         public bool IsOfficial { get; set; }
 
+        [JsonProperty]
         public bool IsMachineWide { get; set; }
 
+        [JsonProperty]
         public bool IsEnabled { get; set; }
 
         public PackageSourceCredential Credentials { get; set; }
 
+        [JsonProperty]
         public string Description { get; set; }
 
+        [JsonProperty]
         public bool IsPersistable { get; private set; }
```

Comment:
I wouldn't be very comfortable with the changes to this file for a final PR.

1. This file is under `src\NuGet.Core`, which means that we [ship it to nuget.org](https://www.nuget.org/packages/NuGet.Configuration). This means I think it should be "done right", because it's an API.
2. It looks arbitrary why some properties get the `JsonProperty` attribute, but others like `MaxHttpRequestPerSource`, `IsOfficial` and `Credentials` do not. This means in a serialization-deserialization round trip it's not going to have the same values, which makes it "feel wrong".
3. Your source branch appears to be old, so missing [`ClientCertificats`](https://github.com/NuGet/NuGet.Client/blob/34ceb7d3d5a70ba4b1ecf196e5902dcc9ffed05d/src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSource.cs#L56). I don't even know what it would mean to serialize and round trip a certificate. Serializing the cert's private key feels like a bad idea, but if it's supposed to be possible to use the deserialized PackageSource to create a NuGet.Protocol.SourceRepository and query the source, then it would be necessary.
4. Since NuGet's config is XML, it feels weird that we're adding `JsonProperty` attributes. Without knowing the context of what's happening in PM UI, it makes it look like we're moving NuGet's config to JSON.

I have no idea how difficult it would be, but if this is all just to make PM UI and/or VS options show the right labels in the UI, then I would prefer if `src\NuGet.Clients` had its own serializable abstraction of a source with just the fields that it needs, rather than modifying the public NuGet SDK.

Reply:
Thats fair and I agree that we should probably not modify core for VS specific things, I can look at creating a VS specific type for this.

Reply:
This afternoon I had the idea that if we have the ability to pass custom JsonConverters, or our own instance of a JsonSerializer, to the service broker, then we can still use `NuGet.Configuration.PackageSource` while still making the serialization specific to VS.

Reply:
Yea I learned a few things from Andrews talk today about brokered services, I was unable to find examples of doing this earlier which is why I went with this approach, hopefully I can get some more clarification about what the best practice is tomorrow from Andrew.

Reply:
Yes, you can certainly provide `JsonConverter` instances to the `JsonSerializer` behind a brokered service See a [sample here](https://dev.azure.com/devdiv/DevDiv/_git/VS?path=/src/env/shell/UIInternal/Packages/OutputChannel/ServiceJsonRpcDescriptorWithCommonConverters.cs).